ttnn/cpp/ttnn/tensor/xtensor/conversion_utils.hpp

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/tensor/shape/small_vector.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include <ttnn/tensor/xtensor/xtensor_all_includes.hpp>
+
+namespace ttnn::experimental::xtensor {
+
+template <typename E>
+ttnn::Shape get_shape_from_xarray(const E& xarr) {
+    ttnn::SmallVector<uint32_t> shape_dims;
+    for (size_t i = 0; i < xarr.shape().size(); ++i) {
+        shape_dims.push_back(xarr.shape()[i]);
+    }
+    return ttnn::Shape(shape_dims);
+}
+
+template <class VectorType = float, DataType TensorType = DataType::BFLOAT16>
```

Comment:
Why have a default template parameter for `TensorType` -- would it be better to keep it explicit, especially if we're dropping bits in the data-type? On second thought, does this make more sense as a function parameter instead?

Reply:
Yeah, these should accept this and a few other params as function arguments. Changing.